internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -1180,6 +1148,46 @@ func (r *ClusterDeploymentsReconciler) createNewCluster(
 		}
 	}
 
+	return clusterParams
+}
+
+func (r *ClusterDeploymentsReconciler) createNewCluster(
+	ctx context.Context,
+	log logrus.FieldLogger,
+	key types.NamespacedName,
+	clusterDeployment *hivev1.ClusterDeployment,
+	clusterInstall *hiveext.AgentClusterInstall) (ctrl.Result, error) {
+
+	log.Infof("Creating a new cluster %s %s", clusterDeployment.Name, clusterDeployment.Namespace)
+
+	pullSecret, err := getPullSecretData(ctx, r.Client, r.APIReader, clusterDeployment.Spec.PullSecretRef, key.Namespace)
+	if err != nil {
+		log.WithError(err).Error("failed to get pull secret")
+		return r.updateStatus(ctx, log, clusterInstall, nil, err)
+	}
+
+	releaseImage, err := r.addReleaseImage(ctx, log, clusterInstall.Spec, pullSecret, nil)
+
+	if err != nil {
+		log.WithError(err)
+		_, _ = r.updateStatus(ctx, log, clusterInstall, nil, err)
+		// The controller will requeue after one minute, giving the user a chance to fix releaseImage
+		return ctrl.Result{Requeue: true, RequeueAfter: longerRequeueAfterOnError}, nil
+	}
+
+	clusterParams := CreateClusterParams(clusterDeployment, clusterInstall, pullSecret, *releaseImage.Version, *releaseImage.CPUArchitecture)
+
+	if clusterInstall.Spec.IgnitionEndpoint != nil {
```

Comment:
a bit strange to leave this part outside of CreateClusterParams 
Maybe it should be another parameter? you can do the parsing outside and pass `ignitionEndpoint` as a parameter as well

Reply:
Good idea. I will parse outside and pass it in as a parameter.